src/main/java/org/aludratest/service/jms/QueueReceiver.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (C) 2015 Hamburg Sud and the contributors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.aludratest.service.jms;
+
+import java.io.Serializable;
+
+import javax.jms.Queue;
+
+import org.aludratest.service.jms.data.TextMessageData;
+import org.databene.commons.Validator;
+
+/**
+ * Receives messages from a JMS {@link Queue}.
+ * @author Volker Bergmann
+ */
```

Comment:
Please add Javadoc comment for generic parameter. Give example how to declare possible subclass.